src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/AvoidDuplicateElementInitialization.cs

```diff
@@ -0,0 +1,133 @@
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftQualityGuidelinesAnalyzersResources.AvoidDuplicateElementInitializationDescription), MicrosoftQualityGuidelinesAnalyzersResources.ResourceManager, typeof(MicrosoftQualityGuidelinesAnalyzersResources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessage,
+                                                                             DiagnosticCategory.Usage,
+                                                                             DiagnosticHelpers.DefaultDiagnosticSeverity,
+                                                                             isEnabledByDefault: true,
+                                                                             description: s_localizableDescription,
+                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        private static readonly ConcurrentDictionary<string, SyntaxGenerator> s_generators = new ConcurrentDictionary<string, SyntaxGenerator>();
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            analysisContext.RegisterOperationAction(AnalyzeOperation, OperationKind.ObjectCreation);
+        }
+
+        private static void AnalyzeOperation(OperationAnalysisContext context)
+        {
+            var objectInitializer = (IObjectCreationOperation)context.Operation;
+            if (objectInitializer.Initializer == null)
+            {
+                return;
+            }
+
+            var initializedElementIndexes = new HashSet<object[]>(ConstantArgumentEqualityComparer.Instance);
+
+            foreach (var initializer in objectInitializer.Initializer.Initializers)
+            {
+                if (initializer is ISimpleAssignmentOperation assignment &&
+                    assignment.Target is IPropertyReferenceOperation propertyReference &&
+                    propertyReference.Arguments.Length != 0)
+                {
+                    var values = GetConstantArgumentValues(propertyReference.Arguments);
+                    if (values != null && !initializedElementIndexes.Add(values))
+                    {
+                        var generator = s_generators.GetOrAdd(
```

Comment:
I am not sure I understand why we need any generators at all here. I think it is ok to just use `value?.ToString() ?? "null"` for the diagnostic text. The diagnostic location will make it clear what syntax needs to be fixed.

Reply:
I meant `string.Join(", ", values.Select(value => value?.ToString() ?? "null"))`

Reply:
As I said, the only thing I was concerned about what having the index expression in the diagnostic match the way the constants would be specified in the target language, but since the only difference really is whether strings are quoted/escaped or not, I guess it's fine. Done.